viewer-prototype/src/browser/trace-viewer/components/xy-output-component.tsx

```diff
@@ -63,15 +63,11 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
 
     renderTree(): React.ReactNode {
         this.onSeriesChecked = this.onSeriesChecked.bind(this);
-        return <React.Fragment>
-            {this.state.XYTree.map(entry => {
-                return <CheckboxComponent key={entry.id}
-                    id={entry.id}
-                    name={entry.labels[0]}
-                    checked={this.state.checkedSeries.find(id => entry.id === id) ? true : false}
-                    onChecked={this.onSeriesChecked} />
-            })}
-        </React.Fragment>;
+        return <XYTree
+            entries = {this.state.XYTree }
+            checkedSeries={this.state.checkedSeries}
```

Comment:
Indeed, if it's part of the state of the caller, it should be in the props of the tree.